evalml/pipelines/component_graph.py

```diff
@@ -378,6 +384,14 @@ def _transform_features(
             dict: Outputs from each component.
         """
         X = infer_feature_types(X)
+        if not fit:
+            if not _schema_is_equal(X.ww.schema, self._input_types):
```

Comment:
@bchen1116 I think we can get rid of `schema_is_equal` and let woodwork do the comparison for us if we use `exclude`. Something like this

```python
        if not fit:
            if X.ww.select(exclude=['integer'], return_schema=True) != self._input_types:
                raise ValueError(
                    "Input X data types are different from the input types the pipeline was fitted on."
                )
        else:
            self._input_types = X.ww.select(exclude=['integer'], return_schema=True)
```

Reply:
Can add a test for this case! I [brought it up](https://ayx-innovation-labs.slack.com/archives/C019J1GSMKR/p1636666403016200) to the woodwork team, but seems like this is something they want to keep

Reply:
I looked into this yesterday and I think it's cause the DateTime format is None in X2 in the repro you shared.

I will try to come up with a minimal repro and share with ww team. This ends up working

```python
from evalml.demos import load_fraud

X, y = load_fraud(1000)

X2 = X.ww.copy()

X.ww.schema == X2.ww.schema
```